README.md
# Device Tree Editor (dtweb)
![Device Tree Editor](https://www.udoo.org/wp-content/uploads/2017/12/dtweb.png)

## Features
This application allows you to configure the pinmuxing of your UDOO board. You can export several devices on the pinout:

 * UART ports
 * CAN interfaces
 * I2C buses
 * SPI interfaces
 * PWM signals
 * ...

## Board support

* UDOO Quad
* UDOO Dual
* UDOO Neo (all editions)


debian/changelog
dtweb (1.15) bionic; urgency=medium

  * Removed gksu/chromium integration. It is now possible to start dtweb from udoo-web-conf.

 -- Francesco Montefoschi <francesco.montefoschi@udoo.org>  Sun, 22 Sep 2019 01:19:12 +0000

dtweb (1.14.1) bionic; urgency=medium

  * dtbkernel4 at 23c6c7b4d23a8eae1740bac0d509bedd72273607

 -- Francesco Montefoschi <francesco.montefoschi@udoo.org>  Sat, 21 Sep 2019 16:01:10 +0200

dtweb (1.14) bionic; urgency=medium

  * UDOO Neo on Bionic support

 -- Francesco Montefoschi <francesco.montefoschi@udoo.org>  Sun, 15 Sep 2019 00:51:26 +0200

dtweb (1.13.2) xenial; urgency=medium

  * Fixed kernel detection under qemu

 -- Francesco Montefoschi <francesco.montefoschi@udoo.org>  Wed, 16 Nov 2016 15:19:48 +0100

dtweb (1.13.1) xenial; urgency=medium

  * Fixed postinst script

 -- Francesco Montefoschi <francesco.montefoschi@udoo.org>  Mon, 14 Nov 2016 12:39:20 +0100

dtweb (1.13) xenial; urgency=medium

  * Enabled SPI1/2 for UDOO Quad/Dual
  * Multicore build proper support
  * Added dtweb-headless

 -- Francesco Montefoschi <francesco.montefoschi@udoo.org>  Fri, 11 Nov 2016 11:33:29 +0100

dtweb (1.12) xenial; urgency=medium

  * Kernel 4 support

 -- Francesco Montefoschi <francesco.montefoschi@udoo.org>  Wed, 30 Aug 2016 18:00:00 +0000

dtweb (1.11.1) xenial; urgency=medium

  * Xenial

 -- Francesco Montefoschi <francesco.montefoschi@udoo.org>  Wed, 24 Aug 2016 16:25:36 +0000

dtweb (1.11) trusty; urgency=medium

  * Added ecspi2

 -- Francesco Montefoschi <francesco.monte@gmail.com>  Wed, 20 Jul 2016 16:05:28 +0100

dtweb (1.10) trusty; urgency=medium

  * Updated device trees to f9d152e66cb598c88860e9713a910f199b6d4d99

 -- Francesco Montefoschi <francesco.monte@gmail.com>  Tue, 14 Mar 2016 14:02:06 +0100

dtweb (1.9) trusty; urgency=medium

  * Release at af66494a7f39b71f9d80871fce5f49b09cbcc71c

 -- Francesco Montefoschi <francesco.monte@gmail.com>  Tue, 16 Feb 2016 16:16:31 +0100

dtweb (1.8) trusty; urgency=medium

  * UDOO Quad/Dual compatibility

 -- Francesco Montefoschi <francesco.monte@gmail.com>  Fri, 08 Jan 2016 15:54:38 +0100

dtweb (1.7) trusty; urgency=medium

  * Major refactoring in utils

 -- Francesco Montefoschi <francesco.monte@gmail.com>  Wed, 30 Dec 2015 16:55:24 +0100

dtweb (1.6) trusty; urgency=medium

  * Enable SPDIF and I2S audio on UDOO Neo

 -- Francesco Montefoschi <francesco.monte@gmail.com>  Wed, 23 Dec 2015 10:34:39 +0100

dtweb (1.5) trusty; urgency=medium

  * getters in udoo*ctl utils

 -- Francesco Montefoschi <francesco.monte@gmail.com>  Thu, 17 Dec 2015 18:40:08 +0100

dtweb (1.4) trusty; urgency=medium

  * Added udoom4ctl

 -- Francesco Montefoschi <francesco.monte@gmail.com>  Thu, 17 Dec 2015 12:45:07 +0100

dtweb (1.3) trusty; urgency=medium

  * Fixes for defconfigs

 -- Francesco Montefoschi <francesco.monte@gmail.com>  Wed, 16 Dec 2015 11:19:45 +0100

dtweb (1.2) trusty; urgency=medium

  * UDOO Neo support

 -- Francesco Montefoschi <francesco.monte@gmail.com>  Fri, 11 Dec 2015 17:50:49 +0100

dtweb (1.1) trusty; urgency=medium

  * Improved GUI, usability and added warning to prevent short circuits between iMX6 and SAM3X.

 -- Francesco Montefoschi <francesco.monte@gmail.com>  Thu, 12 Nov 2015 14:12:56 +0100

dtweb (1.0) trusty; urgency=low

  * Initial release

 -- Francesco Montefoschi <francesco.monte@gmail.com>  Wed, 04 Nov 2015 16:10:50 +0000


public/boards/qdl/board.js
window.board = {
    id: 'qdl',
    name: 'UDOO Quad/Dual',
    minPin: 0,
    maxPin: 55
};

window.board.features = {
    "pwm1": {
        description: 'PWM 1',
        cssClass: 'success',
        configurations: [
            {pins: [8]}
        ]
    },
    "pwm2": {
        description: 'PWM 2',
        cssClass: 'success',
        configurations: [
            {pins: [9]}
        ]
    },
    "pwm3": {
        description: 'PWM 3',
        cssClass: 'success',
        configurations: [
            {pins: [4]}
        ]
    },
    "pwm4": {
        description: 'PWM 4',
        cssClass: 'success',
        configurations: [
            {pins: [5]}
        ]
    },
    
    "usdhc1": {
        description: 'SD card 1',
        cssClass: 'info',
        configurations: [
            {pins: [2, 3, 4, 5, 8, 9]}
        ]
    },
    
    "uart3": {
        description: 'UART 3 at /dev/ttymxc2',
        cssClass: 'warning',
        configurations: [
            {pins: [47, 53]}
        ]
    },
    "uart5": {
        description: 'UART 5 at /dev/ttymxc4',
        cssClass: 'warning',
        configurations: [
            {pins: [16, 17]}
        ]
    },
    
    "i2c1": {
        description: 'I2C bus 1',
        cssClass: 'danger',
        configurations: [
            {pins: [20, 21]}
        ]
    },
    
    "1wire": {
        description: '1-Wire (on any pin)',
        cssClass: 'danger',
        configurations: [
            {pins: [2]},
            {pins: [3]},
            {pins: [4]},
            {pins: [5]},
            {pins: [6]},
            {pins: [7]},
            {pins: [8]},
            {pins: [9]},
            {pins: [10]},
            {pins: [11]},
            {pins: [12]},
            {pins: [13]},
            {pins: [14]},
            {pins: [15]},
            {pins: [16]},
            {pins: [17]},
            {pins: [18]},
            {pins: [19]},
            {pins: [20]},
            {pins: [21]},
            {pins: [22]},
            {pins: [23]},
            {pins: [24]},
            {pins: [25]},
            {pins: [26]},
            {pins: [27]},
            {pins: [28]},
            {pins: [29]},
            {pins: [30]},
            {pins: [31]},
            {pins: [32]},
            {pins: [33]},
            {pins: [34]},
            {pins: [35]},
            {pins: [36]},
            {pins: [37]},
            {pins: [38]},
            {pins: [39]},
            {pins: [40]},
            {pins: [41]},
            {pins: [42]},
            {pins: [43]},
            {pins: [44]},
            {pins: [45]},
            {pins: [46]},
            {pins: [47]},
            {pins: [48]},
            {pins: [49]},
            {pins: [50]},
            {pins: [51]},
            {pins: [52]},
            {pins: [53]}
        ]
    },
    
    "sound_spdif": {
        description: 'SPDIF audio',
        cssClass: 'info',
        depends: [
            'spdif',
        ],
        configurations: [
            {pins: [21, 44]}
        ]
    },
    
    "flexcan1": {
        description: 'CANBUS 1',
        cssClass: 'warning',
        configurations: [
            {pins: [54, 55]}
        ]
    },
    
    "ecspi1": {
        description: 'SPI1',
        cssClass: 'success',
        configurations: [
            {pins: [36, 37, 45, 46]}
        ]
    },
    
    "ecspi2": {
        description: 'SPI2',
        cssClass: 'success',
        configurations: [
            {pins: [34, 50, 51, 52]}
        ]
    }
    
/*
    "SPI5": [
      {pins: [4, 5, 8, 9]}
    ],
    "DIGITALAUDIO": [
      {pins: [29, 30, 32, 33, 34, 35]}
    ],
*/
};
